py-polars/polars/utils/_construction.py

```diff
@@ -819,7 +819,7 @@ def _sequence_of_series_to_pydf(
     data_series: list[PySeries] = []
     for i, s in enumerate(data):
         if not s.name:
-            s.rename(column_names[i], in_place=True)
+            s = s.rename(column_names[i], in_place=False)
```

Comment:
Doesn't this and the `.cast` on line 825 mean that if we have to change the dtype also, we have two copies being made in rapid succession?

Reply:
I see you refering "shallow copies" but when I [drill down into `rename(in_place=False)` I see it calls `self.alias(name)` which [has a `.clone()`](https://github.com/pola-rs/polars/blob/1c25e0dc102430ecee75e66d0f61c0c987617e06/py-polars/polars/series/series.py#LL1898C13-L1898C13).

Reply:
A clone on a series us always shallow.